examples/jsm/loaders/GLTFLoader.js

```diff
@@ -3685,7 +4006,15 @@ class GLTFParser {
 			const input = animationDef.parameters !== undefined ? animationDef.parameters[ sampler.input ] : sampler.input;
 			const output = animationDef.parameters !== undefined ? animationDef.parameters[ sampler.output ] : sampler.output;
 
-			pendingNodes.push( this.getDependency( 'node', name ) );
+			const useExtension = target.extensions && target.extensions[ EXTENSIONS.KHR_ANIMATION_POINTER ];
+
+			let nodeDependency = undefined;
+			if ( useExtension )
+				nodeDependency = this.getAnimationPointerDependency( target );
```

Comment:
It may be good to consider the plugin system for handling a extension. It provides an extensibility while it keeps the simplicity for processing the glTF core in the loader unless the extension handler really needs to be in the loader core. If the current Plugin API doesn't fit to process the extension, please consider to propose a good new API for handling animation extension.

Reply:
Yes, as written in my original message that was the plan.
I now refactored it to be contained entirely in an animation, and added two new callbacks.